src/QMCWaveFunctions/RotationHelper.h

```diff
@@ -0,0 +1,267 @@
+                           const std::vector<size_t>& C2node_dn,
+                           const ValueVector_t& detValues_up,
+                           const ValueVector_t& detValues_dn,
+                           const GradMatrix_t& grads_up,
+                           const GradMatrix_t& grads_dn,
+                           const ValueMatrix_t& lapls_up,
+                           const ValueMatrix_t& lapls_dn,
+                           const ValueMatrix_t& M_up,
+                           const ValueMatrix_t& M_dn,
+                           const ValueMatrix_t& Minv_up,
+                           const ValueMatrix_t& Minv_dn,
+                           const GradMatrix_t& B_grad,
+                           const ValueMatrix_t& B_lapl,
+                           const std::vector<int>& detData_up,
+                           const size_t N1,
+                           const size_t N2,
+                           const size_t NP1,
+                           const size_t NP2,
+                           const std::vector<std::vector<int>>& lookup_tbl);
+
+  //helper function to evaluatederivative; evaluate orbital rotation parameter derivative using table method
+  void table_method_eval(std::vector<ValueType>& dlogpsi,
+                         std::vector<ValueType>& dhpsioverpsi,
+                         const ParticleSet::ParticleLaplacian_t& myL_J,
+                         const ParticleSet::ParticleGradient_t& myG_J,
+                         const size_t nel,
+                         const size_t nmo,
+                         const ValueType& psiCurrent,
+                         const std::vector<RealType>& Coeff,
+                         const std::vector<size_t>& C2node_up,
+                         const std::vector<size_t>& C2node_dn,
+                         const ValueVector_t& detValues_up,
+                         const ValueVector_t& detValues_dn,
+                         const GradMatrix_t& grads_up,
+                         const GradMatrix_t& grads_dn,
+                         const ValueMatrix_t& lapls_up,
+                         const ValueMatrix_t& lapls_dn,
+                         const ValueMatrix_t& M_up,
+                         const ValueMatrix_t& M_dn,
+                         const ValueMatrix_t& Minv_up,
+                         const ValueMatrix_t& Minv_dn,
+                         const GradMatrix_t& B_grad,
+                         const ValueMatrix_t& B_lapl,
+                         const std::vector<int>& detData_up,
+                         const size_t N1,
+                         const size_t N2,
+                         const size_t NP1,
+                         const size_t NP2,
+                         const std::vector<std::vector<int>>& lookup_tbl);
+
+  void checkInVariables(opt_variables_type& active)
+  {
+    if (Optimizable && !IsCloned)
+    {
+      if (myVars.size())
+        active.insertFrom(myVars);
+      Phi->storeParamsBeforeRotation();
```

Comment:
@spinedaflores it becomes clear to me setting deltaP zero should happen in checkInVariables since storeParamsBeforeRotation is called here.